### DISCUSSION TITLE
Where did the default value for max epochs come from?

### DISCUSSION THREAD

**Discussion by wigging** [microsoft/mattergen#199] [Q&A]
In `mattergen/conf/data_module/mp_20.yaml` the default value for `max_epochs` is 900 and for `mattergen/conf/data_module/alex_mp_20.yaml` the default value for `max_epochs` is 2200. Where did these max epoch values come from? Why is max epochs for MP-20 lower than for Alex MP-20?

**Comment by wigging** [comment]
@danielzuegner Do you have any comments regarding this?

**Reply by danielzuegner** [reply]
These came from our early experiments with the datasets, where we found that the MP-20 model was quite converged after 900 epochs, while the Alex-MP-20 model needed more epochs to fully converge.

**Reply by wigging** [reply]
And how did you determine this convergence? Did you look at the loss value or something else?

**Reply by danielzuegner** [reply]
Yes, that's when the validation loss would no longer decrease or even start to go up again.

**Reply by wigging** [reply]
So if I'm training with a new dataset how do I know what value to use for max epochs? Do I just guess what the max epochs should be and keep track of the loss then adjust max epochs accordingly for subsequent training?

**Reply by danielzuegner** [reply]
Hi @wigging,

I recommend using an early stopping callback (see [here](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.EarlyStopping.html#lightning.pytorch.callbacks.EarlyStopping)) with a generous patience to avoid prematurely stopping training. I'd monitor validation loss -- the model will automatically stop training if val loss hasn't decreased in `<patience>` many epochs.

**Reply by wigging** [reply]
Yes, that is what I used in our previous discussion https://github.com/microsoft/mattergen/discussions/183 except I didn't explicitly define a patience value but it looks like the default is 3. But you still have to provide some value for max epochs.

**Reply by danielzuegner** [reply]
Yes, as far as I know you do, but you can set it very high such that it is effectively no constraint on the training length. I think 3 is very low, I would set it to at least 10, perhaps more.
